fv3core/stencils/del2cubed.py

```diff
@@ -73,47 +70,98 @@ def corner_fill(q: FloatField):
             q = q[0, -1, 0]
 
 
-def compute(qdel: FloatField, nmax: int, cd: float, km: int):
-    grid = spec.grid
-    origin = (grid.isd, grid.jsd, 0)
-
-    # Construct some necessary temporary storage objects
-    fx = utils.make_storage_from_shape(
-        qdel.shape, origin=origin, cache_key="del2cubed_fx"
-    )
-    fy = utils.make_storage_from_shape(
-        qdel.shape, origin=origin, cache_key="del2cubed_fy"
-    )
-
-    # set up the temporal loop
-    ntimes = min(3, nmax)
-    for n in range(1, ntimes + 1):
-        nt = ntimes - n
-        origin = (grid.is_ - nt, grid.js - nt, 0)
+class HyperdiffusionDamping:
+    """
+    Fortran name is del2cubed
+    """
+
+    def __init__(self, grid, qdel: FloatField, km: int):
+        """
+        Args:
+            grid: fv3core grid object
+            qdel (inout): Variable to be filterd
+            km: Number of vertical levels (grid.npz)
+        """
+        self.grid = spec.grid
+        self.origin = (self.grid.isd, self.grid.jsd, 0)
+        self.domain = (self.grid.nid, self.grid.njd, km)
```

Comment:
when km is npz, this is usually self.grid.domain_shape_full(), but because it needs to be a custom km, this is perfect as-is.

Reply:
update -- km is always grid.npz so this can be self.grid.domain_shape_full(),